nrf-hal-common/src/i2s.rs

```diff
@@ -341,28 +330,24 @@ impl I2S {
     /// Transmits the given `tx_buffer`.
     /// Returns a value that represents the in-progress DMA transfer.
     #[allow(unused_mut)]
-    pub fn tx<Word, B>(mut self, buffer: B) -> Result<Transfer<B>, Error>
+    pub fn tx<W, B>(mut self, buffer: B) -> Result<Transfer<B>, Error>
     where
-        Word: SupportedWordSize,
-        B: Deref + StableDeref + 'static + I2SBuffer,
-        B::Target: AsRef<[Word]>,
+        B: ReadBuffer<Word = W>,
     {
-        if (buffer.ptr() as usize) < SRAM_LOWER || (buffer.ptr() as usize) > SRAM_UPPER {
-            return Err(Error::DMABufferNotInDataMemory);
-        }
-
-        if buffer.maxcnt() > MAX_DMA_MAXCNT {
+        let (ptr, len) = unsafe { buffer.read_buffer() };
+        let maxcnt = (len / (core::mem::size_of::<u32>() / core::mem::size_of::<W>())) as u32;
+        if maxcnt > MAX_DMA_MAXCNT {
             return Err(Error::BufferTooLong);
         }
+        if (ptr as usize) < SRAM_LOWER || (ptr as usize) > SRAM_UPPER {
```

Comment:
Wouldn't this have to be

```suggestion
        if (ptr as usize) < SRAM_LOWER || (ptr as usize + maxcnt) > SRAM_UPPER {
```

to check for the *whole* buffer, not just its start?

Reply:
Ahh, i was thinking it'd always be located either "inside RAM" or not?

Reply:
Hmm, yeah, that's true